ilastik/experimental/api/_base.py

```diff
@@ -43,7 +43,7 @@ def __init__(self):
             self._predict_op.Image.connect(self._feature_sel_op.OutputImage)
             self._predict_op.LabelsCount.setValue(classifer.label_count)
 
-        def predict(self, data):
+        def predict(self, data, xarray_roi=dict()):
```

Comment:
```python
def predict(self, data, xarray_roi=None):
    if xarray_roi is None:
        xarray_roi = {}
```

Reply:
what is the objection against the default value?

Reply:
Because default function values are evaluated and assigned on function definition, _not_ evaluation.This doesn't play well with mutable values: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments.

Reply:
doesn't happen with `dict()` though...